### DISCUSSION TITLE
[Completed] Progress with Commander XT

### DISCUSSION THREAD

**Discussion by PeiceOfToast** [EvanMulawski/FanControl.CorsairLink#1] [General]
HI,
Big fan of the project and the plugins.
Is there anything the community could do to help?
I've been dying to ditch all the single-party apps and use speed fan for all my temperature control.

**Comment by EvanMulawski** [comment]
Hi @PeiceOfToast - I just received an XT and am starting to implement support for it. I'm planning to have a second beta release by 1/16 with support for it.

**Reply by PeiceOfToast** [reply]
That's awesome.
If you ever have a donation page ill be sure to help out for your time.

**Reply by EvanMulawski** [reply]
@PeiceOfToast Almost ready, just doing testing and cleanup. The only thing that won't work is the automatic pairing of the sensor with the control, so that will need to be done manually. I opened an issue in the FanControl repo to possibly address this in the future (https://github.com/Rem0o/FanControl.Releases/issues/1462).

**Reply by PeiceOfToast** [reply]
Would you happen to know what that process will look like to pair the controller with the fan control?
As simple as adding a device?

**Reply by EvanMulawski** [reply]
Yes, it's two clicks in the UI for each fan channel. You would open the menu for the Control, select "Pair speed sensor...", and then select the corresponding fan.

**Reply by EvanMulawski** [reply]
@PeiceOfToast Released. https://github.com/EvanMulawski/FanControl.CorsairLink/releases/tag/v1.0.0-beta.4

**Reply by PeiceOfToast** [reply]
I'll do some testing and see if I run into any issues tomorrow evening.
Again, Thanks for your help with the project.
Love the open-source community

**Reply by bjongbloed** [reply]
Unfortunately it's not working for me. Here's a bit of logging:

```
20-1-2023 14:48:51: [CorsairLink] DeviceManager: Found 3 Corsair device(s)
  name=LCD Cap for Elite Capellix coolers, devicePath=\\?\hid#vid_1b1c&pid_0c39#7&2544036e&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_01#8&2a7d46db&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

20-1-2023 14:48:51: [CorsairLink] DeviceManager: Found 1 supported Corsair device(s)
  name=CORSAIR iCUE COMMANDER Core, devicePath=\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}

20-1-2023 14:48:51: [CorsairLink] An exception occurred attempting to initialize device 'CORSAIR iCUE COMMANDER Core (205537824242)' (\\?\hid#vid_1b1c&pid_0c32&mi_00#8&12b9ba96&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}):
20-1-2023 14:48:51: [CorsairLink] System.IO.IOException: Operation failed early: The parameter is incorrect ---> System.ComponentModel.Win32Exception: The parameter is incorrect
   --- End of inner exception stack trace ---
   at HidSharp.Platform.Windows.NativeMethods.OverlappedOperation(IntPtr ioHandle, IntPtr eventHandle, Int32 eventTimeout, IntPtr closeEventHandle, Boolean overlapResult, NativeOverlapped* overlapped, UInt32& bytesTransferred)
   at HidSharp.Platform.Windows.WinHidStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at CorsairLink.CommanderCoreDevice.SendCommand(ReadOnlySpan`1 command, ReadOnlySpan`1 data)
   at CorsairLink.CommanderCoreDevice.Initialize()
   at CorsairLink.CommanderCoreDevice.Connect()
   at FanControl.CorsairLink.CorsairLinkPlugin.FanControl.Plugins.IPlugin.Initialize()
```

AC V143, and the beta-4 release
It's the commander that came with the H170i and one that came with the 7000d case.

Maybe I'm doing something wrong? If you need more info, feel free to msg me. Thanks for making this!

**Reply by PeiceOfToast** [reply]
So I suggest you uninstall all corsair controller software. 
If the Corsair control service is running it will fail.

Looking at the case it's only a repeater from your Motherboard to the fans, I would check your bios to see that it's in PWM mode for the header. 
Also, could you verify your fans are PWM controllable? It should be a 4Pin connector.

I've been running on the Core XT, and it auto-picked up all my fans.
It won't auto-detect ranges but I can manually set my RPM range.

**Reply by EvanMulawski** [reply]
@bjongbloed Moved to #6.
